core/src/net/buffer_pool.rs

```diff
@@ -0,0 +1,109 @@
+use crate::net::buffer::{BufferChunk, Chunk, DefaultChunk};
+
+use std::collections::VecDeque;
+
+pub const INITIAL_BUFFER_LEN: usize = 5;
+const MAX_POOL_SIZE: usize = 10000;
+
+/// Methods required by a ChunkAllocator
+pub trait ChunkAllocator: Send + 'static {
+    fn get_chunk(&self) -> Box<dyn Chunk>;
```

Comment:
I would introduce an associated type here and change the `get_chunk` method to return it instead. Something like:
```rust
    type Allocation: DerefMut<Target = dyn Chunk>;
    fn get_chunk(&self) -> Allocation;
```
`Box` is always managed by the global allocator of your binary.

Reply:
... or just return `*mut dyn Chunk` and have
```rust
unsafe fn dealloc(&self, ptr: *mut dyn Chunk);
``` as part of the interface. That wouldn't pollute the rest of the codebase with generics.